lib/Importer/ONNX.cpp

```diff
@@ -243,29 +243,58 @@ bool ONNXModelLoader::loadOperator(const ONNX_NAMESPACE::NodeProto &op) {
     // Pads : {pad_top, pad_left, pad_bottom, pad_right}
     std::vector<unsigned_t> pads = getPads(dict);
 
+    // Load the inputs:
     auto in = getNodeValueOrCreateVariableByName(op.input(0));
-    Tensor *w = getTensorByName(op.input(1));
-
-    // Transpose the weights to the right format. Glow expects to read the
-    // weights in the format CRSK. ONNX stores the operators as KCRS.
-    // C - output_depth, R - filter_height, S - filter_width, K - input_depth.
-    Tensor wtag;
-    w->transpose(&wtag, NCHW2NHWC);
+    Node *filterTransposeNode = NULL;
+    std::vector<unsigned_t> kernelShape(2);
+    size_t depth;
+
+    if (hasTensorWithName(op.input(1))) {
+      // The filter is a constant tensor.
+      Tensor *w = getTensorByName(op.input(1));
+
+      // Transpose the weights to the right format. Glow expects to read the
+      // weights in the format CRSK. ONNX stores the operators as KCRS.
+      // C - output_depth, R - filter_height, S - filter_width, K - input_depth.
+      Tensor wtag;
+      w->transpose(&wtag, NCHW2NHWC);
+
+      // The structure of the conv weigts is: NHWC. We take the C, which is the
+      // number of filters. We use this value to calculate the size of the bias
+      // if it is not specified.
+      depth = wtag.dims()[0];
+
+      // Construct the Filter field.
+      auto *filterVariable =
+          G_.getParent()->createVariable("conv.filter", wtag);
+      filterTransposeNode = filterVariable;
+
+      // Get the kernel shape from the input.
+      kernelShape[0] = filterVariable->dims()[1];
+      kernelShape[1] = filterVariable->dims()[2];
+    } else {
+      // The filter is generated by a node (example with depthwise
+      // convolution). Transpose it to meet the Glow expectation.
+      auto filter_node = getNodeValueOrCreateVariableByName(op.input(1));
+      filterTransposeNode = G_.createTranspose(opName, filter_node, NCHW2NHWC);
 
-    // The structure of the conv weigts is: NHWC. We take the C, which is the
-    // number of filters. We use this value to calculate the size of the bias
-    // if it is not specified.
-    size_t depth = wtag.dims()[0];
+      const NodeValue filterValue = filterTransposeNode->getNthResult(0);
```

Comment:
nit: filterTransposeNode->getResult() should make it.

Reply:
`filterTransposeNode` was declared as `Node` and only has `getNthResult`. Declaring it as `TransposeNode` or with `auto` enables `getResult`.